core/src/main/java/com/scalar/db/common/checker/OperationChecker.java

```diff
@@ -159,6 +163,17 @@ private void check(ScanAll scanAll) throws ExecutionException {
       throw new IllegalArgumentException(
           CoreError.OPERATION_CHECK_ERROR_CROSS_PARTITION_SCAN_ORDERING.buildMessage(scanAll));
     }
+    Optional<Ordering> orderingOnBlobColumn =
+        scanAll.getOrderings().stream()
+            .filter(
+                ordering -> metadata.getColumnDataType(ordering.getColumnName()) == DataType.BLOB)
+            .findFirst();
+    if (orderingOnBlobColumn.isPresent()
+        && new JdbcConfig(config).getJdbcUrl().startsWith("jdbc:db2:")) {
```

Comment:
I'm wondering if there is any other way to determine whether the storage supports BLOB type for ordering. Since `OperationChecker` contains `StorageInfoProvider`, how about adding `boolean StorageInfo.isBlobTypeSupportedForOrdering()` or something? I don't insist on this approach and I'm okay with the current way, though.

Reply:
I also think that whether it should be handled should be the responsibility of storage-side classes.
We also make similar changes for Oracle Database, so having a helper method, like @komamitsu suggested, is more proper.

Reply:
I think, similar to `CosmosOperationChecker` and `DynamoOperationChecker`, we can add a `JdbcOperationChecker` and move the logic into it.

https://github.com/scalar-labs/scalardb/blob/bff508b5eeaf53b1e67232f3598107fe5eedfa91/core/src/main/java/com/scalar/db/storage/cosmos/CosmosOperationChecker.java
https://github.com/scalar-labs/scalardb/blob/bff508b5eeaf53b1e67232f3598107fe5eedfa91/core/src/main/java/com/scalar/db/storage/dynamo/DynamoOperationChecker.java

Reply:
Thanks for the suggestions. 
I followed @brfrn169 idea by moving the checking logic in the `RdbEngineDb2.throwIfCrossPartitionScanOrderingOnBlobColumnNotSupported()` method and have it called by `JdbcOperationChecker`. 
cf. https://github.com/scalar-labs/scalardb/pull/3000/commits/b199319da988c70b427b8f6dcee677df6359ec5b